scripts/deploy.ts

```diff
@@ -1,45 +1,109 @@
+ */
 import { ethers } from 'ethers';
-import path from "path";
-import fs from "fs";
+import fs from 'fs';
+import path from 'path';
+import yargs from 'yargs';
+
 import {
+  ChainMap,
+  IChainConnection,
   MultiProvider,
   chainConnectionConfigs,
   objMap,
   serializeContracts,
 } from '@hyperlane-xyz/sdk';
-import { HypERC20Deployer } from "../src/deploy";  
 
-// Function that takes 2 CLI arguments, the private key and the token config file
-async function deployTradeRoute() {
+import { HypERC20Deployer } from '../src/deploy';
+
+async function deployWarpRoute() {
   const argv = await yargs
-  .option('private-key', {
-    type: 'string',
-    describe: 'Private key for signing transactions',
-    demandOption: true,
-  })
-  .option('config', {
-    type: 'string',
-    describe: 'Path to JSON config file',
-    demandOption: true,
-  })
-  .argv;
-
-  const signer = new ethers.Wallet(argv['private-key']);
-    const config = JSON.parse(fs.readFileSync(path.resolve(argv.config), "utf-8"))
-    const multiProvider = new MultiProvider(
-      objMap(chainConnectionConfigs, (_chain, conf) => ({
-        ...conf,
-        signer: signer.connect(conf.provider),
-      })),
+    .option('private-key', {
+      type: 'string',
+      describe: 'Private key for signing transactions',
+      demandOption: true,
+    })
+    .option('token-config', {
+      type: 'string',
+      describe: 'Path to token config JSON file',
+      demandOption: true,
+    })
+    .option('chain-config', {
+      type: 'string',
+      describe: 'Path to chain config JSON file',
+    }).argv;
+
+  const privateKey = argv['private-key'];
+  const tokenConfigPath = argv['token-config'];
+  const chainConfigPath = argv['chain-config'];
+
+  console.log('Reading warp route configs');
+
+  const tokenConfig = JSON.parse(
+    fs.readFileSync(path.resolve(tokenConfigPath), 'utf-8'),
+  );
+  const targetChains = Object.keys(tokenConfig);
+  console.log(
+    `Found token configs for ${targetChains.length} chains:`,
+    targetChains.join(', '),
+  );
+
+  const chainConfig = chainConfigPath
+    ? JSON.parse(fs.readFileSync(path.resolve(chainConfigPath), 'utf-8'))
+    : null;
+  if (chainConfig) {
+    const customChains = Object.keys(chainConfig);
+    console.log(
+      `Found custom configs for ${customChains.length} chains:`,
+      customChains.join(', '),
     );
+  }
+
+  let multiProviderConfig: ChainMap<any, IChainConnection> = {};
+  for (const chain of targetChains) {
+    if (chainConfig && chainConfig[chain]) {
+      // Use custom config
+      multiProviderConfig[chain] = {
+        provider: new ethers.providers.JsonRpcProvider(
+          chainConfig[chain].rpcUrl,
+          chainConfig[chain].id,
+        ),
+        confirmations: chainConfig[chain].confirmations || 1,
+        blockExplorerUrl: chainConfig[chain].blockExplorerUrl,
```

Comment:
Should we just splat the rest in there? To support the overrides?

Reply:
I didn't use destructuring here because I didn't want the extra fields (rpcUrl, id) to be in the final config object. But I can add a line for overrides, sure.